libsplinter/src/admin/service/builder.rs

```diff
@@ -0,0 +1,232 @@
+        self.key_permission_manager = Some(key_permission_manager);
+        self
+    }
+
+    /// Sets the coordinator timeout for the two-phase commit consensus engine.
+    pub fn with_coordinator_timeout(mut self, coordinator_timeout: Duration) -> Self {
+        self.coordinator_timeout = Some(coordinator_timeout);
+        self
+    }
+
+    /// Sets the routing table writer instance.
+    pub fn with_routing_table_writer(
+        mut self,
+        routing_table_writer: Box<dyn RoutingTableWriter>,
+    ) -> Self {
+        self.routing_table_writer = Some(routing_table_writer);
+        self
+    }
+
+    /// Constructs the AdminServce.
+    ///
+    /// # Errors
+    ///
+    /// Returna an [InvalidStateError] if any required properties are missing.
+    pub fn build(self) -> Result<super::AdminService, InvalidStateError> {
+        let coordinator_timeout = self
+            .coordinator_timeout
+            .unwrap_or_else(|| Duration::from_secs(DEFAULT_COORDINATOR_TIMEOUT));
+
+        let orchestrator = self.orchestrator.ok_or_else(|| {
+            InvalidStateError::with_message(
+                "An admin service requires a service_orchestrator".into(),
+            )
+        })?;
+        let orchestrator = Arc::new(Mutex::new(orchestrator));
+
+        let node_id = self.node_id.ok_or_else(|| {
+            InvalidStateError::with_message("An admin service requires a node_id".into())
+        })?;
+
+        #[cfg(feature = "service-arg-validation")]
+        let service_arg_validators = self.service_arg_validators;
+
+        let admin_store = self.admin_store.ok_or_else(|| {
+            InvalidStateError::with_message(
+                "An admin service requires an admin_service_store".into(),
+            )
+        })?;
+
+        let peer_connector = self.peer_connector.ok_or_else(|| {
+            InvalidStateError::with_message(
+                "An admin service requires a peer_manager_connector".into(),
+            )
+        })?;
+
+        let signature_verifier = self.signature_verifier.ok_or_else(|| {
+            InvalidStateError::with_message("An admin service requires a signature_verifier".into())
+        })?;
+
+        let key_verifier = self.key_verifier.ok_or_else(|| {
+            InvalidStateError::with_message(
+                "An admin service requires an admin_key_verifier".into(),
+            )
+        })?;
+        let key_permission_manager = self.key_permission_manager.ok_or_else(|| {
+            InvalidStateError::with_message(
+                "An admin service requires a key_permission_manager".into(),
+            )
+        })?;
+
+        let routing_table_writer = self.routing_table_writer.ok_or_else(|| {
+            InvalidStateError::with_message(
+                "An admin service requires an routing_table_writer".into(),
+            )
+        })?;
+
+        #[cfg(feature = "admin-service-event-store")]
+        let admin_event_store = admin_store.clone_boxed();
```

Comment:
The event store needs to be able to be a different type in the case that the admin store is set to YAML. The builder should have a separate with_admin_event_store function until we make YamlAdminServiceStore read only.
This will require updating the depreacted new method and the daemon to still pass it in.

Reply:
Fixed